### PR TITLE
[Docs] Add high-level `ROADMAP.md`

### PR DESCRIPTION
**Pull request by foundrytom**
The TAC checklist shamed me

### CODE REVIEW

**Review by elliotcmorris — APPROVED** [review]
TAC checklist aside, i'm personally glad to see this. I didn't know myself our 2024 goals to this level of specificity.

File: ROADMAP.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+## Y24 H1: OpenAssetIO UI v1.0.0a1
+
+An initial release of the UI delegation layer that sits on top of
+OpenAssetIO, allowing custom Manager UI elements to be directly embedded
+into Host application.
+
+Initially limited to the Qt UI framework.
+
+## Y24 onwards
```
Comment (feltech) [inline ROADMAP.md:52]: [condier] I guess C API is super-deprioritised?

File: ROADMAP.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+## Y24 Q1: v1.0.0
+
+This release will provide additional introspection methods and
+debugging functionality, along with language parity between Hosts and
+Managers.
+
+**Hosts:** C++, Python
+**Managers:** C++, Python
+
+- [ ] Auth related error codes.
```
Comment (feltech) [inline ROADMAP.md:36]: [consider] We also have a few places where we throw std exceptions (or let std exceptions propagate from library calls), when we should create our own exception types.  

Perhaps this is covered by "Sundry minor API Change issues" above (I'd argue that adding proper exceptions should go in the beta).

File: ROADMAP.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+- [ ] Debug trace logging support.
+- [ ] Entity introspection API methods.
```
[the remark this answers is not in the document]
Comment (feltech) [reply ROADMAP.md:38]: Ah, like

- #31 ?

File: ROADMAP.md [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+- [ ] C++ Plugin System
+- [ ] Hybrid C++/Python manager bridge.
+- [ ] Out-of-process Python
+- [ ] Migrate landing page/examples to [OpenAssetIO-MediaCreation](https://github.com/OpenAssetIO/OpenAssetIO-MediaCreation)
+
+## Y24 H1: OpenAssetIO UI v1.0.0a1
```
Comment (feltech) [inline ROADMAP.md:44]: [consider] 

- #7

**Review by feltech — COMMENTED** [review]
Looks good. I flicked through the open issues to see if anything stood out, but looks mostly covered (pending questions/comments below).

File: ROADMAP.md (hunk 2 above)
Comment (foundrytom) [reply ROADMAP.md:36]: I was lumping this more in the "to C++ migration" category, but we can call more of these things out if you like?

File: ROADMAP.md (hunk 4 above)
Comment (foundrytom) [reply ROADMAP.md:44]: This was another we were waiting for real-world use cases for before it was prioritised as its incredibly hypothetical at present.

File: ROADMAP.md (hunk 2 above)
Comment (feltech) [reply ROADMAP.md:36]: I suppose calling out specifically about exception types is a bit low level for this doc, but perhaps something along the lines of "mapping out and concretely defining failure modes" perhaps?

File: ROADMAP.md (hunk 4 above)
Comment (feltech) [reply ROADMAP.md:44]: We could take a look at what few limited integrations currently exist and we have visibility of the code (Ayon, Shotgrid) to see what, if any, SDK access they require beyond what OpenAssetIO can/will provide, and get an idea how much priority they should be given. 

Also ASWF/studio contacts could be asked (though I suspect limited help there).

Or maybe its a case of we accept that SDK access will be required, but that's not the responsibility of OpenAssetIO, and at best OpenAssetIO will help inform the context (node name, etc) to make bespoke SDK usage a little bit easier.

Comment (foundrytom) [reply ROADMAP.md:44]: Yeah 100% Just trying to balance hours in the day and functionality that is currently blocking in-application workflows... trying not to have too many things in flight at once.

File: ROADMAP.md (hunk 2 above)
Comment (foundrytom) [reply ROADMAP.md:36]: Yeah thats a good idea

**Pull request merged**